mypy 1.5.0 errors: "`Unsupported operand types`"

New error from new `mypy` version (1.5.0) for `beta` branch); the code works more or less fine as before:
```bash
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

holidays\holiday_base.py:402: error: Unsupported operand types for == ("Mapping" and "HolidayBase")  [operator]
Found 1 error in 1 file (checked 314 source files)
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -35,7 +35,7 @@ repos:
         exclude: ^(docs)
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v1.4.1
+    rev: v1.5.0
     hooks:
       - id: mypy
         additional_dependencies: [types-all]

--- a/holidays/holiday_base.py
+++ b/holidays/holiday_base.py
@@ -399,7 +399,7 @@ class HolidayBase(Dict[date, str]):
             if getattr(self, attribute_name, None) != getattr(other, attribute_name, None):
                 return False
 
-        return dict.__eq__(self, other)
+        return dict.__eq__(cast("Mapping[Any, Any]", self), other)
 
     def __getattr__(self, name):
         try:

